libfreerdp/core/update.c

```diff
@@ -1555,45 +1736,65 @@ static void update_write_pointer_color(wStream* s, POINTER_COLOR_UPDATE* pointer
 		Stream_Write(s, pointer_color->andMaskData, pointer_color->lengthAndMask);
 
 	Stream_Write_UINT8(s, 0); /* pad (1 byte) */
+	return TRUE;
 }
 
-static void update_send_pointer_color(rdpContext* context, POINTER_COLOR_UPDATE* pointer_color)
+static BOOL update_send_pointer_color(rdpContext* context, POINTER_COLOR_UPDATE* pointer_color)
 {
 	wStream* s;
 	rdpRdp* rdp = context->rdp;
+	BOOL ret = FALSE;
 
 	s = fastpath_update_pdu_init(rdp->fastpath);
-        update_write_pointer_color(s, pointer_color);
-	fastpath_send_update_pdu(rdp->fastpath, FASTPATH_UPDATETYPE_COLOR, s, FALSE);
+	if (!s)
+		return FALSE;
+
+	if (!update_write_pointer_color(s, pointer_color))
+		goto out_fail;
+
+	ret = fastpath_send_update_pdu(rdp->fastpath, FASTPATH_UPDATETYPE_COLOR, s, FALSE);
+
+out_fail:
 	Stream_Release(s);
+	return ret;
 }
 
-static void update_send_pointer_new(rdpContext* context, POINTER_NEW_UPDATE* pointer_new)
+static BOOL update_send_pointer_new(rdpContext* context, POINTER_NEW_UPDATE* pointer_new)
 {
 	wStream* s;
 	rdpRdp* rdp = context->rdp;
+	BOOL ret = FALSE;
 
 	s = fastpath_update_pdu_init(rdp->fastpath);
+	if (!s)
+		return FALSE;
 
-	Stream_EnsureRemainingCapacity(s, 16);
+	if (!Stream_EnsureRemainingCapacity(s, 16))
+		goto out_fail;
 
 	Stream_Write_UINT16(s, pointer_new->xorBpp); /* xorBpp (2 bytes) */
         update_write_pointer_color(s, &pointer_new->colorPtrAttr);
 
-	fastpath_send_update_pdu(rdp->fastpath, FASTPATH_UPDATETYPE_POINTER, s, FALSE);
+	ret = fastpath_send_update_pdu(rdp->fastpath, FASTPATH_UPDATETYPE_POINTER, s, FALSE);
 
+out_fail:
 	Stream_Release(s);
+	return ret;
 }
 
-static void update_send_pointer_cached(rdpContext* context, POINTER_CACHED_UPDATE* pointer_cached)
+static BOOL update_send_pointer_cached(rdpContext* context, POINTER_CACHED_UPDATE* pointer_cached)
 {
 	wStream* s;
 	rdpRdp* rdp = context->rdp;
+	BOOL ret;
 
 	s = fastpath_update_pdu_init(rdp->fastpath);
+	if (!s)
```

Comment:
I don't think so. Capacity check is done in  update_write_pointer_color and fastpath_update_pdu_init initializes the stream with FASTPATH_MAX_PACKET_SIZE.